tests/Mindee.UnitTests/Extraction/ImageExtractorTest.cs

```diff
@@ -0,0 +1,128 @@
+using Mindee.Extraction;
+using Mindee.Input;
+using Mindee.Parsing.Common;
+using Mindee.Product.BarcodeReader;
+using Mindee.Product.MultiReceiptsDetector;
+
+namespace Mindee.UnitTests.Extraction
+{
+    [Trait("Category", "ImageExtractor")]
+    public class ImageExtractorTest
+    {
+        [Fact]
+        public async Task GivenAnImage_ShouldExtractPositionFields()
+        {
+            var image = new LocalInputSource("Resources/products/multi_receipts_detector/default_sample.jpg");
+            var response = await GetMultiReceiptsDetectorPrediction("complete");
+            var inference = response?.Document.Inference;
+
+            var extractor = new ImageExtractor(image);
+            Assert.Equal(1, extractor.GetPageCount());
+
+
+            if (inference != null)
+            {
+                foreach (var page in inference.Pages)
+                {
+                    var subImages = extractor.ExtractImagesFromPage(page.Prediction.Receipts, page.Id);
+                    for (int i = 0; i < subImages.Count; i++)
+                    {
+                        var extractedImage = subImages[i];
+                        Assert.NotNull(extractedImage.Image);
+                        extractedImage.WriteToFile("Resources/output/");
+
+                        var source = extractedImage.AsInputSource();
+                        Assert.Equal(
+                            $"default_sample_page-001_{i + 1:D3}.jpg",
+                            source.Filename
+                        );
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public async Task GivenAnImage_ShouldExtractValueFields()
+        {
+            var image = new LocalInputSource("Resources/products/barcode_reader/default_sample.jpg");
+            var response = await GetBarcodeReaderPrediction("complete");
+            var inference = response?.Document.Inference;
+
+            var extractor = new ImageExtractor(image);
+            Assert.Equal(1, extractor.GetPageCount());
+
+            if (inference != null)
+            {
+                foreach (var page in inference.Pages)
+                {
+                    var codes1D = extractor.ExtractImagesFromPage(page.Prediction.Codes1D, page.Id, "barcodes_1D.png");
+                    for (int i = 0; i < codes1D.Count; i++)
+                    {
+                        var extractedImage = codes1D[i];
+                        Assert.NotNull(extractedImage.Image);
+                        var source = extractedImage.AsInputSource();
+                        Assert.Equal(
+                            $"barcodes_1D_page-001_{i + 1:D3}.jpg",
```

Comment:
The output format should be the same across all libs, either change Java or change Python.

Reply:
I think I'll change python & also take the opportunity to update indexes on there.